app/controllers/application_controller.rb

```diff
@@ -42,20 +42,23 @@ def verify_lock
     end
 
     def set_locale
-      if params[:locale] && I18n.available_locales.include?(params[:locale].to_sym)
-        session[:locale] = params[:locale]
-      end
+      I18n.locale = current_locale
```

Comment:
Setting `I18n.locale` first and then updating everything else depending on `I18n.locale` makes a lot of sense to me :+1:.